File: Python/35_SearchInsertPosition.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        s = 0
        e = len(nums)-1

        while s<=e:
            m = (s+e) // 2
            if nums[m] == target:
                return m
            elif nums[m] < target:
                s = m + 1
            else:
                e = m - 1

        return s
        

File: C++/35_SearchInsertPosition.cpp
using namespace std;
#include <vector>

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l=0;
        int r=nums.size()-1;

        if (nums[nums.size()-1] < target) return nums.size();
        
        while (l<r) {
            int m = (r+l)/2;
            if (nums[m]==target) return m;
            else if(nums[m] < target) l = m + 1;
            else r = m;
        }

        return l;
    }
};